### PR TITLE
Add bad day report controller

### PR DESCRIPTION
**Pull request by tylerwill3000**

### CODE REVIEW

File: OddJava/src/main/java/com/oddball/challenges/BadDayController.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import java.time.LocalDate;
+import java.util.List;
+
+@RestController
+@RequestMapping("/bad-day")
+@RequiredArgsConstructor
+public class BadDayController {
+    private final BadDayService badDayService;
+
+    @GetMapping("/weeks/{weekToCheck}")
```
Comment (tylerwill3000) [inline OddJava/src/main/java/com/oddball/challenges/BadDayController.java:15]: Implements requirement #1 from https://github.com/tylerwill3000/challenges/tree/master/OddJava#bad-day-reports

File: OddJava/src/main/java/com/oddball/challenges/BadDayController.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    public List<BadWeek> getBadWeeks(@PathVariable LocalDate weekToCheck) {
+        return badDayService.getBadWeeks(weekToCheck);
+    }
+
+    @GetMapping("/streaks")
```
Comment (tylerwill3000) [inline OddJava/src/main/java/com/oddball/challenges/BadDayController.java:20]: Implements requirement #2 from https://github.com/tylerwill3000/challenges/tree/master/OddJava#bad-day-reports

File: OddJava/src/main/java/com/oddball/challenges/BadDayService.java [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+    private Map<Long, User> getUsersById(Set<Long> userIds) {
+        return userRepository.findAllByIdIn(userIds)
+            .stream()
+            .collect(toMap(User::getId, identity()));
+    }
+
+    /**
+     * Get all bad days (from mood and stress) grouped by user for a given date range inclusive
+     * @param from Starting day (inclusive) to retrieve bad days for. Null = no lower bound
```
Comment (tylerwill3000) [inline OddJava/src/main/java/com/oddball/challenges/BadDayService.java:109]: The ability to pass `null` to exclude the boundaries for pulling bad days allows us to use this method for both reports without duplicating any code or business logic that classifies what a 'bad day' is

File: OddJava/src/main/java/com/oddball/challenges/stress/StressRepository.java [hunk 4]
```diff
@@ -1,10 +1,26 @@
...
+import java.util.Date;
+import java.util.List;
+
 @Repository
 public interface StressRepository extends CrudRepository<Stress, Long> {
+    @Query("""
+        select new com.oddball.challenges.BadDayDto(s.userId, s.date)
+        from Stress s
+        left join Mood m on m.userId=s.userId and m.date=s.date
+        where (s.stress in (4, 5) and (m is null or m.mood=3))
```
Comment (tylerwill3000) [inline OddJava/src/main/java/com/oddball/challenges/stress/StressRepository.java:18]: This condition covers the following requirement:

"A user's mood is a 3, but their stress for that same day is a 4 or 5"

The condition works properly since the absence of a mood for this day (`m.null` in the query) is the same as a neutral mood (3), which qualifies for a bad stress day.

File: OddJava/src/test/groovy/com/oddball/challenges/BadDayServiceSpec.groovy
Comment (tylerwill3000) [inline OddJava/src/test/groovy/com/oddball/challenges/BadDayServiceSpec.groovy:1]: These tests are not exhaustive of all edge cases, but they demonstrate that the 2 primary report generating functions return the proper data in order for relatively interesting datasets. These tests identified at least 2 bugs as I was completing the 2 reports.
